### PR TITLE
client/rpcserver: Add wallet routes

### PR DESCRIPTION
**Pull request by JoeGruffins**
Add init, newwallet, openwallet, and wallets routes.

~Change command arguments to a slice of strings.~ <- Implemented in another pr.

### CODE REVIEW

File: client/rpcserver/handlers.go [hunk 1]
```diff
@@ -9,13 +9,27 @@ import (
...
+	openWalletRoute  = "openwallet"
 	preRegisterRoute = "preregister"
+	versionRoute     = "version"
+	walletsRoute     = "wallets"
+)
+
+const (
+	INITIALIZED     = "initialized"
+	WALLET_LOCKED   = "wallet locked"
+	WALLET_UNLOCKED = "wallet unlocked"
```
Comment (buck54321) [inline client/rpcserver/handlers.go:32]: While not explicitly forbidden, upper-case constants are not common in go, and underscores [are discouraged](https://golang.org/doc/effective_go.html#mixed-caps).

File: client/rpcserver/handlers.go [hunk 2]
```diff
@@ -132,31 +254,96 @@ func sortHelpKeys() []string {
...
+   	existed.
+      "islocked" (bool): Whether the wallet is locked.
+    }`,
+	},
+	openWalletRoute: {`assetID "appPass"`,
+		`Open an existing wallet.
+
+Args:
+    assetID (int): The asset's SLIP-0044 ID, 42 for dcr or 0 for btc.
+    appPass (string): The Dex Client password.
```
Comment (buck54321) [inline client/rpcserver/handlers.go:316]: ~~`Dex Client`~~ -> `DEX client`, or `dex client`

File: client/rpcserver/handlers.go [hunk 3]
```diff
@@ -132,31 +254,96 @@ func sortHelpKeys() []string {
...
-		"fee" (float): The dex registration fee.
-	}`,
+    obj: The wallets result.
+    [
+      {
+        "symbol" (string): The coin symbol.
+        "assetID" (int): The coin SLIP-0044 number.
+        "open" (bool): Whether the wallet is unlocked.
+        "running" (bool): Whether the wallet is running.
+        "updated" (int): ???
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/rpcserver/handlers.go:341]: Oh yeah. I meant to ask directly then forgot. Sorry.

File: client/rpcserver/handlers.go [hunk 4]
```diff
@@ -20,20 +35,334 @@ func createResponse(op string, res interface{}, resErr *msgjson.Error) *msgjson.
...
+		}
+	}
 	return createResponse(req.Route, &res, nil)
 }
 
-// handleVersion responds to the version request.
+// handleInit handles requests for init.
+func handleInit(s *RPCServer, req *msgjson.Message) *msgjson.ResponsePayload {
+	reqPayload := ""
+	err := json.Unmarshal(req.Payload, &reqPayload)
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/rpcserver/handlers.go]: Changed all instances to use this pattern.

File: client/rpcserver/handlers.go [hunk 5]
```diff
@@ -20,20 +35,334 @@ func createResponse(op string, res interface{}, resErr *msgjson.Error) *msgjson.
...
+	if err != nil {
+		resErr := &msgjson.Error{Code: msgjson.RPCParseError, Message: "unable to unmarshal request"}
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/rpcserver/handlers.go:87]: Changed in all places.

File: client/rpcserver/handlers.go [hunk 6]
```diff
@@ -70,8 +107,93 @@ func handleVersion(s *RPCServer, req *msgjson.Message) *msgjson.ResponsePayload
...
+		resErr := &msgjson.Error{Code: msgjson.RPCErrorUnspecified, Message: errMsg}
+		return createResponse(req.Route, nil, resErr)
+	}
+	s.notifyWalletUpdate(form.AssetID)
+	res.IsNew = true
+	res.IsLocked = false
+	return createResponse(req.Route, res, nil)
+}
+
+// handleOpenWallet handles requests for openWallet. *msgjson.ResponsePayload.Error
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/rpcserver/handlers.go:150]: All comments in handlers.go now 80 columns or less now.

**Comment by JoeGruffins** [comment]
Will push one more commit for msgjson types and erroring on new wallet when a wallet exists.

**Comment by JoeGruffins** [comment]
> Tested well.
> 
> How hard would it be to have **dexcctl** prompt for `appPass` without echo for `init`, `newwallet`, etc.?

Could you give me some time to think about that? This is something supported by dcrd's ctl either.
I use `export HISTCONTROL=ignorespace` in my .bashrc, and just use a space whenever I go to unlock my wallet, that way my pass isn't stored in history. But I guess someone could be watching over my shoulder though...

File: client/rpcserver/types.go [hunk 7]
```diff
@@ -32,6 +33,27 @@ type preRegisterResponse struct {
 	Fee uint64 `json:"fee"`
 }
 
+// newWalletResponse is a response to newWalletRequest.
+type newWalletResponse struct {
+	IsNew    bool `json:"isnew"`
+	IsLocked bool `json:"islocked"`
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/rpcserver/types.go:39]: I meant to remove this. A wallet that exist but is not new will now return an error.

File: client/rpcserver/handlers.go [hunk 8]
```diff
@@ -96,7 +234,8 @@ func ListCommands() string {
 	var sb strings.Builder
 	var err error
 	for _, r := range sortHelpKeys() {
-		_, err = sb.WriteString(fmt.Sprintf("%s %s\n", r, helpMsgs[r][0]))
+		_, err = sb.WriteString(fmt.Sprintf("%s %s\n",
+			r, helpMsgs[r][0]))
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply client/rpcserver/handlers.go:238]: Ok. I will stick to <= 80 columns for comments and <= 100 for code from now on. If I recall from our discussion on this in dcrstakepool, it was from the beginning of the code/comment, not the line that we are talking about, correct?

**Comment by JoeGruffins** [comment]
I think I have a simple answer for a password prompt but would like to push it in a separate pr.

File: client/rpcserver/handlers.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (chappjc) [reply client/rpcserver/handlers.go:238]: Maybe I misstated something in the past, but it's absolute columns in the editor window.  But right, I only care about comment wrapping... and extremely long code lines too, but far less than comments, which wrap easily without sacrificing code readability.

**Pull request merged**